tsdb/shard.go

```diff
@@ -452,6 +452,10 @@ func (s *Shard) validateSeriesAndFields(points []models.Point) ([]*FieldCreate,
 		key := string(p.Key())
 		ss := s.index.Series(key)
 		if ss == nil {
+			if len(s.index.series)-1 > s.options.Config.MaxSeriesPerDatabase {
```

Comment:
I think this should be `s.options.Config.MaxSeriesPerDatabase > 0 && ...` so that a setting of `0` disables the limit.